src/phpDebug.ts

```diff
@@ -893,88 +889,52 @@ class PhpDebugSession extends vscode.DebugSession {
         response: VSCodeDebugProtocol.ContinueResponse,
         args: VSCodeDebugProtocol.ContinueArguments
     ) {
-        let xdebugResponse: xdebug.StatusResponse | undefined
-        try {
-            const connection = this._connections.get(args.threadId)
-            if (!connection) {
-                throw new Error('Unknown thread ID ' + args.threadId)
-            }
-            xdebugResponse = await connection.sendRunCommand()
-        } catch (error) {
-            this.sendErrorResponse(response, error)
-            if (xdebugResponse) {
-                this._checkStatus(xdebugResponse)
-            }
-            return
-        }
         response.body = {
             allThreadsContinued: false,
         }
+        const connection = this._connections.get(args.threadId)
+        if (!connection) {
+            this.sendErrorResponse(response, new Error('Unkown thread ID ' + args.threadId))
+            return
+        }
         this.sendResponse(response)
-        this._checkStatus(xdebugResponse)
+        this._checkStatus(await connection.sendRunCommand())
```

Comment:
The problem with this is that you are not handling the error anymore. The response to this command is the only chance to report that the run command failed. XDebug doesn't return a response to the `continue` command unless the command failed or the program stopped again.

If the only way is to send the response straight away, it needs to at least log errors, or only send the response after a timeout.

Reply:
@felixfbecker 

> If the only way is to send the response straight away, it needs to at least log errors, or only send the response after a timeout.

The docs say to send the response that says the action was started, so yes it needs to send the response right away.  As I understand it, it should use `sendErrorResponse()` only if there is some reason it cannot start the action.  The results of the action are sent up the line to vscode in other ways already, these are the different outcomes I can think of:

* success, now it is paused again - we let xdebug know in `_checkStatus()`
* Some exception / error like fatal error - this is also sort of success, the code was run and it is now paused on some problem in the code, so we let vscode know about it in `_checkStatus()`
* The script finishes.  We let vscode know about that on https://github.com/felixfbecker/vscode-php-debug/blob/cf7ad01339f5ea053cc5117f101d6db085112492/src/phpDebug.ts#L281
* There is some connection level error, we let vscode know about those here: 
 https://github.com/felixfbecker/vscode-php-debug/blob/cf7ad01339f5ea053cc5117f101d6db085112492/src/phpDebug.ts#L277-L280
* There is a server level error, we let vscode know about those here:  https://github.com/felixfbecker/vscode-php-debug/blob/cf7ad01339f5ea053cc5117f101d6db085112492/src/phpDebug.ts#L309-L312

It looked like the try/catch was specifically for handling when the connection was not found with that `threadId`, and when that is the case, we now handle it directly by responding with `sendErrorResponse()`.  As far as I could tell it would never throw an exception as part of the call to `connection.sendRunCommand()`, instead it would either return a response with an error in it (handled by `_checkStatus()`, or emit an event which the `phpDebug` is already handling on it's own as noted above.

All vscode needs to know here is that we started running the command, after that point if there is an error we let vscode know about it in other emitted events.

Reply:
Sorry for the late reply. I agree with what you say, but that doesn't change the fact that `sendRunCommand()` can reject with an exception, and that exception needs to be handled _somehow_. Even if it is handled async and we send the response before, it needs an error handler that at a minimum needs to send an error log to VS Code.

Reply:
Ah, I see what you mean, if the xdebug socket is not writable it will reject so need to catch that.

If you are in agreement about using async calls (with addition of handling the reject error) let me know.  If interested I'd be happy to rebase (if possible after a year) or just redo this PR when I have some time.

Reply:
Yes that sounds good